### PR TITLE
Blur previous element when interacting with Cesium canvas.

### PR DESCRIPTION
**Pull request by mramato**
Interacting with a canvas does not automatically blur the previously focused element. This leads to unexpected interaction if the last element was an input field. For example, clicking the mouse wheel could lead to the value in the field changing unexpectedly. The solution is to blur whatever has focus as soon as canvas interaction begins.

@emackey can you see any flaws/problems in this approach?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @mramato!
* :heavy_check_mark: Signed CLA found.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by emackey** [comment]
This seems like a good idea.

One concern I might have is that there might not be a way to get the canvas to receive keyboard input, like hotkeys, if any apps are using that.  Clicking the canvas will cause it to blur itself.

Not sure the solution... maybe blur the active element and then attempt to focus the canvas?  Or just double-check that it's not the canvas that's being blurred?

**Comment by mramato** [comment]
> Not sure the solution... maybe blur the active element and then attempt to focus the canvas? Or just double-check that it's not the canvas that's being blurred?

Good thinking.  However I'm having a hard time actually triggering what I agree would be a bug in the fix here.

I'm using this demo to test: http://localhost:8080/Apps/Sandcastle/index.html?src=HeadingPitchRoll.html

It listens to the document element, but with my code in this branch it still works.  I assume this is because the event bubbles up to the document when no element has focus.  This demo is also kind of sloppy by listening to document because if you active the geocoder and press up/down, it still affects the plane instead of the geocoder.

Working on that assumption, I then changed the Sandcastle to use `viewer.scene.canvas.addEventListener('keydown', function(e) {` so that it now only listens to the canvas.  I then select the geocoder input field and nothing happens with up/down arrows, as to be expected. But, when I click and interact with the Cesium window and then use the arrows, they still work!  As you pointed out, this doesn't seem to be possible because the canvas should no longer be the active element.

So it appears that nothing extra is actually needed here, but I don't fully understand what.

Thoughts?

**Comment by emackey** [comment]
Yeah that does seem to work.  The demo's click handler is registered after the viewer is constructed (of course, because prior to that there's no canvas).  So I guess the blur happens before the focus, and everything is OK.

I still feel like it would be good to check that the canvas isn't blurring itself.  But I don't quite know how an app would manage to get into that state exactly.

**Comment by emackey** [comment]
So yeah, if the canvas does `focus` itself but it has a tab index, then you can actually TAB your way into it if you try hard enough, and then a click will blur it.  Kind of an extreme edge case, probably no one has ever done that before.

**Comment by emackey** [comment]
[yep I went there](http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/only-took-4-years/Apps/Sandcastle/index.html#c=vVnbbuO4GX4VIjdWGll2DnPYjCeoJ8lmZ5vdMZLsblEEGNASHROhSYGi4qjF3G5vd9uH6G2foY8yT9L/p0gdbMc5NKkxQCTyPx8+8tfcUE1uOJszTd4TyebkkGU8n0U/27WgE9vXQyUN5ZLpTkj+dikJ/LKpykUylHxGDSP7xOicXcovm+8u5aW8AakxhT8ZSC3FR+U7bJcPUcbM0BjNx7lhQcfQMZcJuwUFnX5n8x3p9cAclrCEGEXS3JCJivOMKEnMlDnhlSyaJMc3TJpTnhkmrd2Cx9cgbJLL2HAlg01veK/nmKzAAA22Vi+tlm5kMQisvbCv1WZKzXSkMo4a2vE7p7NUsMRvjrRKmTaFFYycYCw3xQj4a9l2jTPrTuCsTb30/ZaysNyVdIax7+BWx63hM6w5fpequctQWK8KRpMLbvn7jWWjKRdu/XVzY84TK3i7uahZpkTuDNxubGBVaE4FLDeiMlKiEFBHJ0LNf3AUVWgaFuPvCmhGCgsTDIx2w/auocB093ashMIdp/cQX6PR8PT45NPp0fGPZ5+Oavovm+Xzl4XynTFNm+WL7y57MfSDVkLYrrElAYWhGZPnKY3ZoSU9rGgcE9L2vQDI9WLLHVJt4InK3Ub5TdMzkNEm/A4yx+XViJt4irtVUQExlVdsDTVuV+RZCg0GxNveqoQJQ8+AnEpsXCcAEjWNjHLrwW7Ur81L6+Jf8iKaaDU7YlcQlyzobu8A45u9vdfb34Rkby/qv+rvvum/Dsmrfr9fiqxs+pnFRq2PDtJO+C1LvtUQ7QsNlk2UntV2VEtZJFRMRUmnvq14ThAoKOgJOlJp7B/SmbPMdBreCSrZSPMZ+HjDqlSnfqXsVB8mBeGzPp8IM/HVnGtsgU4U9eBfCQpH1NDeDKmzXsk75No9fYbH6EqMfTNbMkiA5rd1NTdcmy6UQtPBwCcndFUUegHHQvA0UzyJfjk5f7sXrorkpreAA8TnsxFQiHP+V4SF7Z232CplnNoxijQYVEBLz3jGIkBqGVQIbH1pwDAZCVoQKhMilEoJhTKn9jgB4oxQQ6ZUTLq2IBrBiGiMioYVJeZgCE3QgI9ZLgxPBXfg8KqJeqiqiqTNWSXqFPais+PR8fDCIULpojP3L0rN8DCyZpSrWKM7UZ/8gdgmmdHb0storHKJiTlPp0xDVCBLeRY6UIEiyTMD6iWjGlWUiOXhInIhR31HcKBRGWPtadACvjjyujehNPYi53HxoRgpLo2zolE9IVlpWIlDocMjb4vFkrKy7oSBnV3XtZ4hxRq8k7y70yB3OBXVOtzTwr4Xaf8u7GmHdBiWVyDbh7GMMKT5emgC755jqs76NReHa1Ykai6bVwdWFW02txYFLAKyQwjoZvOcjWnGyF5/v30U8QnSZ1M+MX9iRYvB/6C4SixGzcvbHqerGuuWXRGSbR/R6iwjgCrsDh1lOFfrKBHCRbz7vnUUvFsmR6daLAPSbSb+4pdPn0cfVzq7pG2rXTMl6wqdXxZcbb+OAXmu3y0kY/ft/5CMPL0vFVwGW1s+Ff3+E5KxSsdicB6bigPy1Ex0XzAT3zwpE4iIRPOrqbkzTpbkUWGyHE+JkmV8apDWFYPJtbzHTY+Vj/LUMz3FWc/7kkXx5ulFIdjEkBwmJrG+MrqProwBwaPqIaWw9VKlgM7dWwndp1TCg5xr1NoLpD5hEwrXlf1VU5fTDGMUjhQJzOIzOJ2jK2aOBcPHD8XHJOg4so6fBCx83cdkiSoWTOB9HEjTaU0m93FYoooFh4EPbMplso6npvKjR+NTAwwb7Kc0gVl5+a5SXVAsZUgMTO31VaU1SC0PZ/WN8RymI6qDZZKffvx48fnPoTv0ejj5N8X64255BrzrVrowK7Supw3BYSWyUtH49IFxKAcpb/L3uQB7jzBGUs2DzWX+/+fgFJK73aznCWzNOvFRPGXxNUtajbl65HjA5X9dmBf2HjQO3HFlb2HFClJ/e29eMVonwD039QV0WGiKMyYTeH9cUzTgJeISyL+7+OF0aWbx3yzaHm7Cjq2RoLp1V7DzcGGWZYUoD0cPl4QcKwRVMNWSZFfbxDauG+HGIDOFYAd1Yv7IZ6nSBj9dBFHUMwx6DXor641zqFETxVnmlQ16TeZBwm8IT95fbix8Nb7cILGgWQY7k1zYzwiXGweDHtAvsQplo/3phmlBCyCrDRtMtw9Oy+0oigY9ePVmrJJklBJjqtsiDB0LVlnD5USNoCFEi6gkHKukOFg+6wZGH6w+PAcmORhMd4gNCQYBPzruF0wINQf5F4pMRGG/XtsmhC7Db9TkckNJUXTB2uvuXrdgVGcQLzpWN0CCX0fwKwqBis8ycjH8gH7vQOxA2QrjeiutW2+z+0C4TwYZnm42dq7mbZZw9eA//35OlV9//Q1hDa85va+//gOf7e33OXVYaG85ZXvv5Vz6Hd3IU3Don/iEc/ZzKsB+b7mDCy+ZoC3y9e//qnNUvj57ms4RmVp+lV/8Ksdmvey581S6Ajnyg77N2cLqc+dvohAI8OykXMearrrhV/Rc4n952XDUtwSABVOkFlrwujBWt0uw1bBwjeELUNdbxDpYQZhcga7/BQ)

**Comment by mramato** [comment]
Thanks for your diligence here @emackey, I pushed a fix to check if canvas is the element focused before blurring.

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 Change Log
 ==========
 
+### 1.68.0 - 2020-05-01
```
Comment (emackey) [inline CHANGES.md]: May already!  What happened to April?

**Pull request merged**